src/Icicle/Avalanche/Check.hs

```diff
@@ -88,7 +88,15 @@ checkStatement frag ctx stmt
                    $ checkExp frag (ctxExp ctx) x
                Let n x' <$> go stmts
 
-        ForeachInts n from to stmts
+        While t n from to stmts
+         -> do from' <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) from
+               to'   <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) to
+
+               While t n from' to' <$> go stmts
```

Comment:
yeah, when I actually use this construct I just init and modify the acc manually